projects/plugins/jetpack/extensions/blocks/one-payment/edit.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * External dependencies
+ */
+import { get, map } from 'lodash';
+import classnames from 'classnames';
+import { __ } from '@wordpress/i18n';
+import { useEffect } from '@wordpress/element';
+import { compose, withInstanceId } from '@wordpress/compose';
+import { createBlock, registerBlockVariation } from '@wordpress/blocks';
+import { withDispatch, withSelect } from '@wordpress/data';
+import {
+	InnerBlocks,
+	__experimentalBlockVariationPicker as BlockVariationPicker,
+} from '@wordpress/block-editor';
+
+/**
+ * Internal dependencies
+ */
+import defaultVariations from './variations';
+
+export function JetpackOnePaymentEdit( {
+	setAttributes,
+	hasInnerBlocks,
+	replaceBlock,
+	selectBlock,
+	clientId,
+	className,
+	blockType,
+	variations,
+	defaultVariation,
+} ) {
+	const formClassnames = classnames( className, 'jetpackone-payment' );
+
+	const createBlocksFromInnerBlocksTemplate = innerBlocksTemplate => {
+		const blocks = map( innerBlocksTemplate, ( [ name, attr, innerBlocks = [] ] ) =>
+			createBlock( name, attr, createBlocksFromInnerBlocksTemplate( innerBlocks ) )
+		);
+
+		return blocks;
+	};
+
+	const setVariation = variation => {
+		if ( variation.attributes ) {
+			setAttributes( variation.attributes );
+		}
+
+		if ( variation.innerBlocks ) {
+			replaceBlock( clientId, createBlocksFromInnerBlocksTemplate( variation.innerBlocks ) );
+		}
+
+		selectBlock( clientId );
+	};
+
+	useEffect( () => {
+		// Populate default variation on older versions of WP or GB that don't support variations.
+		if ( ! hasInnerBlocks && ! registerBlockVariation ) {
+			setVariation( defaultVariations[ 0 ] );
+		}
+	} );
+
+	const renderVariationPicker = () => {
+		return (
+			<div className={ formClassnames }>
+				<BlockVariationPicker
+					icon={ get( blockType, [ 'icon', 'src' ] ) }
+					label={ get( blockType, [ 'title' ] ) }
+					instructions={ __( "Please select which kind of payment you'd like to add.", 'jetpack' ) }
```

Comment:
Nitpick: The message uses double quotes while the domain uses single quotes.

Reply:
The message has a single quote inside (`you'd`), so it's ok to use doubles instead of escaping.